tests/rptest/tests/self_test_test.py

```diff
@@ -233,3 +233,66 @@ def test_self_test_cancellable(self):
                 # cancelled before it even had a chance to start, resulting in
                 # a 0 value for duration
                 assert report['duration'] >= 0
+
+    @cluster(num_nodes=3)
+    def test_self_test_unknown_test_type(self):
+        """Assert the self test still runs when an invalid test type is passed.
+           This helps ensure the self test will still work in a cluster
+           with mixed versions of Redpanda."""
+        num_nodes = 3
+
+        self.start_redpanda(num_nodes=num_nodes,
+                            si_settings=SISettings(
+                                test_context=self.test_context,
+                                cloud_storage_enable_remote_read=True,
+                                cloud_storage_enable_remote_write=True))
+
+        #Attempt to run with an unknown test type "pandatest"
+        #and possibly unknown "cloud" test.
+        #The rest of the tests should proceed as normal.
+        request_json = {
+            'tests': [{
+                'type': 'pandatest'
+            }, {
+                'type': 'disk'
+            }, {
+                'type': 'network'
+            }, {
+                'type': 'cloud'
+            }]
+        }
+
+        #Manually invoke self test admin endpoint.
+        self.redpanda._admin._request('POST',
+                                      'debug/self_test/start',
+                                      json=request_json)
+
+        #Populate list of unknown reports.
+        unknown_report_types = ['pandatest']
+        redpanda_versions = [
+            self.redpanda.get_version_int_tuple(node)
+            for node in self.redpanda.nodes
+        ]
+
+        #All nodes should have the same version of
+        #Redpanda running.
+        assert len(set(redpanda_versions)) == 1
+
+        #Cloudcheck was introduced in 24.2.1.
+        #Expect that it will be unknown to nodes running
+        #earlier versions of redpanda.
+        if redpanda_versions[0] < (24, 2, 1):
+            unknown_report_types.append('cloud')
```

Comment:
Do you mean for this to be an upgrade test? Check out usages of RedpandaInstaller if you're trying to write a mixed-version test

Reply:
No, I intended for this to be a single version test. This line is here because this PR is selected to be backported to `24.1.x`.